text/0056-sequence-number-panorama.md

```diff
@@ -0,0 +1,153 @@
+# Sequence number panoramas
+
+## Summary
+
+[summary]: #summary
+
+CEP PR: [casperlabs/ceps#0056](https://github.com/casperlabs/ceps/pull/0056)
+
+We propose citing units by sequence number instead of by hash.
+This will reduce the protocol overhead and bandwidth requirement.
+
+
+## Motivation
+
+[motivation]: #motivation
+
+32-byte hashes of earlier units make up the largest part of Highway messages.
+Replacing them by 8-byte sequence numbers will therefore considerably reduce message sizes,
+and improve performance.
+
+
+## Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+Each unit — the main type of message in the Highway protocol — contains a list of hashes of earlier units.
+That way the sender attests to which units they had received before they created the new one.
+This is what defines the directed acyclic graph (DAG) structure of Highway's protocol state:
+The graph has an arrow `u → v` if `u` _directly cites_ `v`, i.e. the message `u` mentions `v` by hash.
+
+What matters to the protocol's logic is not the graph itself but the partial order it defines:
+If `u → v → w` that means the creator of `u` must also have known about `w`, even if `u` doesn't directly cite `w`.
+We write `u > v` if `v` is reachable from `u` by following any number of arrows.
+
+In particular, the actual message `u` that gets sent over the wire does not need to _directly_ cite every `v < u`.
+In the current implementation, `u` contains a _panorama_, which is a list containing for each (honest) validator the hash of the latest `v < u` created by that validator.
```

Comment:
```suggestion
In the current implementation, `u` contains a _panorama_, which is a list containing for each (non-faulty) validator the hash of the latest `v < u` created by that validator.
```

Reply:
Done in https://github.com/casper-network/ceps/pull/56/commits/365531c9b716b67564fa5c7a74ff6698d4833178.